packages/core/src/shader/Shader.ts

```diff
@@ -46,30 +59,50 @@ export class Shader {
   static create(name: string, subShaders: SubShader[]): Shader;
 
   static create(
-    name: string,
-    vertexSourceOrShaderPassesOrSubShaders: SubShader[] | ShaderPass[] | string,
+    nameOrGalaceanShader: string,
+    vertexSourceOrShaderPassesOrSubShaders?: SubShader[] | ShaderPass[] | string,
     fragmentSource?: string
   ): Shader {
-    const shaderMap = Shader._shaderMap;
-    if (shaderMap[name]) {
-      throw `Shader named "${name}" already exists.`;
-    }
     let shader: Shader;
-    if (typeof vertexSourceOrShaderPassesOrSubShaders === "string") {
-      const shaderPass = new ShaderPass(vertexSourceOrShaderPassesOrSubShaders, fragmentSource);
-      shader = new Shader(name, [new SubShader("Default", [shaderPass])]);
+    const shaderMap = Shader._shaderMap;
+
+    if (!vertexSourceOrShaderPassesOrSubShaders) {
+      if (!this._shaderLab) {
+        throw "ShaderLab not been setted up";
+      }
+      // TODO: render state
+      const shaderInfo = this._shaderLab.parseShader(nameOrGalaceanShader);
+      const subShaderList = shaderInfo.subShaders.map((subshader) => {
+        const passList = subshader.passes.map((pass) => {
+          return new ShaderPass(pass.vert, pass.frag, pass.tags);
+        });
+        return new SubShader(subshader.name, passList, subshader.tags);
+      });
+
+      return new Shader(shaderInfo.name, subShaderList);
     } else {
-      if (vertexSourceOrShaderPassesOrSubShaders.length > 0) {
-        if (vertexSourceOrShaderPassesOrSubShaders[0].constructor === ShaderPass) {
-          shader = new Shader(name, [new SubShader("Default", <ShaderPass[]>vertexSourceOrShaderPassesOrSubShaders)]);
+      if (shaderMap[nameOrGalaceanShader]) {
+        throw `Shader named "${nameOrGalaceanShader}" already exists.`;
+      }
+      if (typeof vertexSourceOrShaderPassesOrSubShaders === "string") {
+        const shaderPass = new ShaderPass(vertexSourceOrShaderPassesOrSubShaders, fragmentSource);
+        shader = new Shader(nameOrGalaceanShader, [new SubShader("Default", [shaderPass])]);
+      } else {
+        if (vertexSourceOrShaderPassesOrSubShaders.length > 0) {
+          if (vertexSourceOrShaderPassesOrSubShaders[0].constructor === ShaderPass) {
+            shader = new Shader(nameOrGalaceanShader, [
+              new SubShader("Default", <ShaderPass[]>vertexSourceOrShaderPassesOrSubShaders)
+            ]);
+          } else {
+            shader = new Shader(nameOrGalaceanShader, <SubShader[]>vertexSourceOrShaderPassesOrSubShaders.slice());
+          }
         } else {
-          shader = new Shader(name, <SubShader[]>vertexSourceOrShaderPassesOrSubShaders.slice());
+          throw "SubShader or ShaderPass count must large than 0.";
         }
-      } else {
-        throw "SubShader or ShaderPass count must large than 0.";
       }
     }
-    shaderMap[name] = shader;
+
+    shaderMap[nameOrGalaceanShader] = shader;
     return shader;
   }
 
```

Comment:
ShaderFactory file should also be migrated to your AST parsing in the future, otherwise it will be a waste of performance.